### PR TITLE
Allow to delete routes

### PR DESCRIPTION
**Pull request by juanfont**
This PR adds support to delete routes from the CLI and the API.

It also deletes the routes of a machine when it is deleted (https://github.com/juanfont/headscale/issues/1228)

### CODE REVIEW

File: machine.go [hunk 1]
```diff
@@ -535,6 +535,11 @@ func (h *Headscale) RefreshMachine(machine *Machine, expiry time.Time) error {
 
 // DeleteMachine softs deletes a Machine from the database.
```
Comment (kradalby) [inline machine.go:536]: What is the difference between hard and soft? why do we have them?

File: machine.go [hunk 2]
```diff
@@ -535,6 +535,11 @@ func (h *Headscale) RefreshMachine(machine *Machine, expiry time.Time) error {
...
 func (h *Headscale) DeleteMachine(machine *Machine) error {
+	err := h.DeleteMachineRoutes(machine)
```
Comment (kradalby) [inline machine.go:538]: Cant this be done by cascading database deletes?

File: machine.go (hunk 1 above)
Comment (juanfont) [reply machine.go:536]: Hard actually deletes the entry in the DB, soft just sets the deleted_at field. We do hard when deleting ephemeral nodes.

File: machine.go (hunk 2 above)
Comment (juanfont) [reply machine.go:538]: If we would use cascade, we would be using soft deleted in the routes https://stackoverflow.com/a/67751408 and I wanted to hard delete them.

**Review by kradalby — APPROVED** [review]

**Pull request merged**